### DISCUSSION TITLE
Rest API Comapactability For Fetching data

### DISCUSSION THREAD

**Discussion by AjinThankachan** [apache/skywalking#8372] [Q&A] [answered]
I am doing some R&D in getting data regarding SKYWALKING like tracing, Events and Metrics. Would like to improve it . So working using python. Is there anyway that i can get the data in JSON or using any api. Else can you tell me how can I populate the data if I need to add more dashboard fileds or make any change in the existing view or add something new as a page in the skywalking. I refered the document .https://github.com/apache/skywalking/blob/master/docs/en/protocols/HTTP-API-Protocol.md

which tells we can get the trace details if we pass the trace id and all details .Instead of that is there any way we can get it bypassing the date time alone which will give all the info in the time span? 
Or do i need to fetch from the back end storage to get it and populate the same on the dashboard. Can you please give a guidence. I am really interested to work on this.

**Comment by wu-sheng** [comment, marked as answer]
We have provided a GraphQL grammar-based query protocol for years. 

https://skywalking.apache.org/docs/main/v8.9.1/en/protocols/query-protocol/#query-protocol

**Reply by AjinThankachan** [reply]
Thanks a lot

**Comment by wu-sheng** [comment]
> I refered the document .https://github.com/apache/skywalking/blob/master/docs/en/protocols/HTTP-API-Protocol.md

This is data report protocol, which doesn't relate to any data visualized on the UI.
